examples/post-sync/main.go

```diff
@@ -0,0 +1,352 @@
+	if errMessage == "" {
+		return nil
+	}
+	if c.statusTracker.IsLogged(rsyncID, commit, errMessage) {
+		return nil
+	}
+	c.log.Info("Sync error detected", "sync", rsyncID.Name, "namespace", rsyncID.Namespace, "kind", rsyncID.Kind, "commit", commit, "status", "failed", "generation", generation, "observedGeneration", observedGeneration, "error", errMessage)
+
+	c.statusTracker.MarkLogged(rsyncID, commit, errMessage)
+	return nil
+}
+
+func (c *SyncStatusController) processRootSyncConditions(conditions []v1beta1.RootSyncCondition) (string, string) {
+	var errorMessages []string
+	var commit string
+
+	for _, condition := range conditions {
+		isErrorCondition := false
+		var conditionType v1beta1.RootSyncConditionType
+
+		switch condition.Type {
+		case v1beta1.RootSyncStalled:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+		case v1beta1.RootSyncReconcilerFinalizerFailure:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+			// Other condition types (Reconciling, Syncing, ReconcilerFinalizing)
+			// are not treated as primary error indicators here for aggregation.
+			// Stalled usually captures persistent source/rendering issues,
+			// and Sync errors are handled via status.errors.
+		}
+
+		if isErrorCondition {
+			errorMessages = append(errorMessages,
+				fmt.Sprintf("%s: %s", conditionType, condition.Message))
+			if commit == "" && condition.Commit != "" {
+				commit = condition.Commit
+			}
+		}
+	}
+
+	return strings.Join(errorMessages, "\n"), commit
+}
+
+func (c *SyncStatusController) processRepoSyncConditions(conditions []v1beta1.RepoSyncCondition) (string, string) {
+	var errorMessages []string
+	var commit string
+
+	for _, condition := range conditions {
+		isErrorCondition := false
+		var conditionType v1beta1.RepoSyncConditionType
+
+		switch condition.Type {
+		case v1beta1.RepoSyncStalled:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+		case v1beta1.RepoSyncReconcilerFinalizerFailure:
+			if condition.Status == metav1.ConditionTrue {
+				isErrorCondition = true
+				conditionType = condition.Type
+			}
+			// Other condition types (Reconciling, Syncing, ReconcilerFinalizing)
+			// are not treated as primary error indicators here for aggregation.
+		}
+
+		if isErrorCondition {
+			errorMessages = append(errorMessages,
+				fmt.Sprintf("%s: %s", conditionType, condition.Message))
+			if commit == "" && condition.Commit != "" {
+				commit = condition.Commit
+			}
+		}
+	}
+
+	return strings.Join(errorMessages, "\n"), commit
+}
+
+// isSuccessfulSync checks if the rsync status no errors in rsync status and all commits match.
+func isSuccessfulSync(status v1beta1.Status) bool {
```

Comment:
should this also test `status.observedGeneration == metadata.generation` and/or `status.lastSyncedCommit == status.Source.Commit`

Reply:
Added to the sync successful check.